packages/core/src/routes/interaction/verifications/user-identity-verification.ts

```diff
@@ -0,0 +1,128 @@
+import { deduplicate } from '@silverhand/essentials';
+import type { Provider } from 'oidc-provider';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import { findSocialRelatedUser } from '#src/lib/social.js';
+import assertThat from '#src/utils/assert-that.js';
+import { maskUserInfo } from '#src/utils/format.js';
+
+import type {
+  SocialIdentifier,
+  VerifiedEmailIdentifier,
+  VerifiedPhoneIdentifier,
+  PreAccountVerifiedInteractionResult,
+  AccountVerifiedInteractionResult,
+  Identifier,
+  InteractionContext,
+} from '../types/index.js';
+import findUserByIdentifier from '../utils/find-user-by-identifier.js';
+import { isProfileIdentifier } from '../utils/index.js';
+import {
+  storeInteractionResult,
+  isAccountVerifiedInteractionResult,
+} from '../utils/interaction.js';
+
+const identifyUserByVerifiedPasscodeIdentifier = async (
+  identifier: VerifiedEmailIdentifier | VerifiedPhoneIdentifier
+) => {
+  const user = await findUserByIdentifier(
+    identifier.key === 'emailVerified' ? { email: identifier.value } : { phone: identifier.value }
+  );
+
+  assertThat(user, new RequestError({ code: 'user.user_not_exist', status: 404 }));
+
+  const { id, isSuspended } = user;
+
+  assertThat(!isSuspended, new RequestError({ code: 'user.suspended', status: 401 }));
+
+  return id;
+};
+
+const identifyUserBySocialIdentifier = async (identifier: SocialIdentifier) => {
+  const { connectorId, userInfo } = identifier;
+
+  const user = await findUserByIdentifier({ connectorId, userInfo });
+
+  if (!user) {
+    const relatedInfo = await findSocialRelatedUser(userInfo);
+
+    throw new RequestError(
+      {
+        code: 'user.identity_not_exists',
+        status: 422,
+      },
+      relatedInfo && { relatedUser: maskUserInfo(relatedInfo[0]) }
+    );
+  }
+
+  const { id, isSuspended } = user;
+
+  assertThat(!isSuspended, new RequestError({ code: 'user.suspended', status: 401 }));
+
+  return id;
+};
+
+const identifyUser = async (identifier: Identifier) => {
+  if (identifier.key === 'social') {
+    return identifyUserBySocialIdentifier(identifier);
+  }
+
+  if (identifier.key === 'accountId') {
+    return identifier.value;
+  }
+
+  return identifyUserByVerifiedPasscodeIdentifier(identifier);
+};
+
+export default async function userIdentityVerification(
```

Comment:
I think `userIdentity` sounds related to social identities, and thus a bit confusing to me.

Reply:
verifyUserAccount?